pkg/network/tracer/tracer_usm_linux_test.go

```diff
@@ -363,6 +372,151 @@ func testTLSClassification(t *testing.T, tr *Tracer, clientHost, targetHost, ser
 	})
 }
 
+func testTLSAMQPClassification(t *testing.T, tr *Tracer, _, targetHost, serverHost string) {
+	skipFunc := composeSkips(skipIfNotLinux, skipIfUsingNAT, skipIfHTTPSNotSupported)
+	skipFunc(t, testContext{
+		serverAddress: serverHost,
+		serverPort:    amqpsPort,
+		targetAddress: targetHost,
+	})
+
+	// Start the proxy server.
+	authority := net.JoinHostPort(serverHost, amqpsPort)
+	cmd, cancel := proxy.NewExternalUnixTransparentProxyServer(t, amqpUnixPath, authority, amqp.AmqpTLS)
+	t.Cleanup(cancel)
+	require.NoError(t, proxy.WaitForConnectionReady(amqpUnixPath))
+	utils.WaitForProgramsToBeTraced(t, "go-tls", cmd.Process.Pid)
+
+	clientDialFn := func() (io.ReadWriteCloser, error) {
+		return net.Dial("unix", amqpUnixPath)
+	}
+
+	amqpTeardown := func(t *testing.T, ctx testContext) {
+		if client, ok := ctx.extras["client"].(*amqp.Client); ok {
+			defer client.Terminate()
+
+			require.NoError(t, client.DeleteQueues())
+		}
+	}
+
+	// Setting one instance of amqp server for all tests.
+	serverAddress := net.JoinHostPort(serverHost, amqpsPort)
+	targetAddress := net.JoinHostPort(targetHost, amqpsPort)
+	require.NoError(t, amqp.RunServer(t, serverHost, amqpsPort, amqp.AmqpTLS))
+
+	tests := []protocolClassificationAttributes{
+		{
+			name: "connect",
+			context: testContext{
+				serverPort:    amqpsPort,
+				serverAddress: serverAddress,
+				targetAddress: targetAddress,
+				extras:        make(map[string]interface{}),
+			},
+			postTracerSetup: func(t *testing.T, ctx testContext) {
+				client, err := amqp.NewClient(amqp.Options{
+					ServerAddress: ctx.serverAddress,
+					WithTLS:       true,
+					TLSDialFn:     clientDialFn,
+				})
+				require.NoError(t, err)
+				ctx.extras["client"] = client
+			},
+			teardown:   amqpTeardown,
+			validation: validateProtocolConnection(&protocols.Stack{Encryption: protocols.TLS, Application: protocols.AMQP}),
+		},
+		{
+			name: "declare channel",
+			context: testContext{
+				serverPort:    amqpsPort,
+				serverAddress: serverAddress,
+				targetAddress: targetAddress,
+				extras:        make(map[string]interface{}),
+			},
+			preTracerSetup: func(t *testing.T, ctx testContext) {
+				detachPID(t, cmd.Process.Pid)
+				client, err := amqp.NewClient(amqp.Options{
+					ServerAddress: ctx.serverAddress,
+					WithTLS:       true,
+					TLSDialFn:     clientDialFn,
+				})
+				require.NoError(t, err)
+				ctx.extras["client"] = client
+				attachPID(t, cmd.Process.Pid)
+			},
+			postTracerSetup: func(t *testing.T, ctx testContext) {
+				client := ctx.extras["client"].(*amqp.Client)
+				require.NoError(t, client.DeclareQueue("test", client.PublishChannel))
+			},
+			teardown:   amqpTeardown,
+			validation: validateProtocolConnection(&protocols.Stack{Encryption: protocols.TLS}),
```

Comment:
nit: add a comment in line 429 "declare channel is not supported, thus we expect to classify the protocol as TLS only"